metrics/inspector_prioritization.py

```diff
@@ -0,0 +1,274 @@
+import boto3
+import pandas as pd
+import numpy as np
+from arcgis.gis import GIS
+from arcgis.features import FeatureLayer
+from arcgis.geometry.filters import intersects
+from sodapy import Socrata
+
+import os
+import logging
+
+from config import (
+    SEGMENTS_FEATURE_SERVICE_URL,
+    DAPCZ_FEATURE_SERVICE_URL,
+    ROW_INSPECTOR_SERVICE_URL,
+)
+from utils import get_logger, s3_csv_to_df, df_to_socrata_dataset
+
+# AWS Credentials
+AWS_ACCESS_ID = os.getenv("EXEC_DASH_ACCESS_ID")
+AWS_PASS = os.getenv("EXEC_DASH_PASS")
+BUCKET = os.getenv("BUCKET_NAME")
+
+# Socrata Credentials
+SO_WEB = os.getenv("SO_WEB")
+SO_TOKEN = os.getenv("SO_TOKEN")
+SO_KEY = os.getenv("SO_KEY")
+SO_SECRET = os.getenv("SO_SECRET")
+DATASET = os.getenv("PRIORITY_DATASET")
+
+# AGOL Credentials
+AGOL_USERNAME = os.getenv("AGOL_USERNAME")
+AGOL_PASSWORD = os.getenv("AGOL_PASSWORD")
+
+PERMITS_FILE = "row_inspector_permit_list.csv"
+SEGMENTS_FILE = "row_inspector_segment_list.csv"
+
+
+def number_of_segments_scoring(permits, segments):
+    """
+    10 points for permits with more than 1 segment, 5 points otherwise.
+
+    Returns
+    -------
+    permits - with a count_segment_scoring column with the scores based on the number of segments
+
+    """
+
+    counts = segments.groupby("FOLDERRSN")["PROPERTYRSN"].count()
+    counts = counts.rename("count_segments")
+    permits = permits.merge(counts, left_on="FOLDERRSN", right_index=True, how="left")
+    permits["count_segments"] = permits["count_segments"].fillna(0)
+
+    # 10 points for permits with more than 1 segment, 5 points otherwise:
+    permits["count_segment_scoring"] = np.where(permits["count_segments"] > 1, 10, 5)
+    return permits
+
+
+def duration_scoring(row):
+    """
+
+    Parameters
+    ----------
+    row - works row-wise on the permits data
+
+    Returns
+    -------
+    scoring based on the number of days the permit is active
+
+    """
+    # each permit type uses a different column from the query for the duration (in days).
+    if row["FOLDERTYPE"] == "RW":
+        duration_column = "TOTAL_DAYS"
+    if row["FOLDERTYPE"] == "DS":
+        duration_column = "WZ_DURATION"
+    if row["FOLDERTYPE"] == "EX":
+        start = pd.to_datetime(row["START_DATE"])
+        end = pd.to_datetime(row["END_DATE"])
+        delta = end - start
+        row["duration"] = delta.days
+        duration_column = "duration"
+
+    # Scoring based on number of days the permit is active:
+    if row[duration_column] <= 6:
+        return 10
+    elif row[duration_column] <= 15:
+        return 5
+    elif row[duration_column] <= 30:
+        return 3
+    return 1
+
+
+def batch_list(data, batch_size=100):
+    for i in range(0, len(data), batch_size):
+        yield data[i : i + batch_size]
+
+
+def retrieve_road_segment_data(segments, gis):
```

Comment:
would be nice to have a little more documentation here. looks like you're grabbing the geometry of every street segment that has a permit so that you can assign it ROW inspector zones.